File: Python/VendingMachine/Domain/hotDrink.py
from Domain.product import Product


class HotDrink(Product):
    def __init__(self, price, place, name, id):
        super().__init__(price, place, name, id)
        self.temperature = 0.0

    def __init__(self, price, place, name, id, temperature):
        super().__init__(price, place, name, id)
        self.temperature = temperature

    def __str__(self):
        return super().__str__() + "\nvolume=" + str(self.temperature)

File: Java/VendingMachine/src/domain/HotDrink.java
package domain;

public class HotDrink extends Product{
    private int temperature;

    @Override
    public String toString() {
        return "HotDrink{" +
                super.toString() + " " +
                "temperature=" + temperature +
                '}';
    }

    public int getTemperature() {
        return temperature;
    }

    public void setTemperature(int temperature) {
        this.temperature = temperature;
    }

    public HotDrink(long id, String name, double price, int place, int temperature) {
        super(id, name, price, place);
        this.temperature = temperature;
    }
}
